selenium_tests/homepage/test_footer.py

```diff
@@ -0,0 +1,78 @@
+from wagtail.core.models import Site
+from wagtail_factories import SiteFactory
+
+from selenium_tests.base import BaseSeleniumTests
+from home.factories import (
+    FooterIndexPageFactory,
+    HomePageFactory,
+    SectionFactory,
+    SectionIndexFactory,
+)
+
+class FooterSeleniumTests(BaseSeleniumTests):
+
+    def setUp(self):
+        Site.objects.all().delete()
+        self.home = HomePageFactory()
+        self.site = SiteFactory(
+            site_name='IoGT',
+            hostname=self.host,
+            port=self.port,
+            is_default_site=True,
+            root_page=self.home
+        )
```

Comment:
I don't yet have a very clear idea of how to organise the creation of test data. In any case, I think it would be best to address this in a future PR - perhaps you can create one for your mixin idea?